ui/components/src/model/units/bytes.ts

```diff
@@ -31,16 +34,37 @@ export const BYTES_UNIT_CONFIG: Readonly<Record<BytesUnitKind, UnitConfig>> = {
 };
 
 // https://stackoverflow.com/questions/15900485/correct-way-to-convert-size-in-bytes-to-kb-mb-gb-in-javascript/18650828#18650828
-export function formatBytes(bytes: number, decimals = 2) {
+export function formatBytes(bytes: number, unitOptions: BytesUnitOptions) {
+  // default to full 'Bytes' formatting
+  const options = unitOptions.abbreviate === undefined ? { ...unitOptions, abbreviate: false } : unitOptions;
+
   if (bytes === 0) return '0 Bytes';
 
+  let decimals = options.decimal_places ?? DEFAULT_DECIMAL_PLACES;
+  // avoids minimumFractionDigits value is out of range error
+  if (decimals < 0) {
+    decimals = 0;
+  } else if (decimals > 100) {
+    decimals = 100;
```

Comment:
Just curious, is `100` just an arbitrary number or is that a real limitation of some code in here?

Reply:
So at one point, I saw this [limit of 100](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed#exceptions) for toFixed (or else a RangeError is thrown), but turns out the limit [should actually be 20](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#minimumfractiondigits) since I switched to use Intl.NumberFormat so I'll update this line. This avoids highly unlikely edge case of:
<img width="355" alt="image" src="https://user-images.githubusercontent.com/9369625/206564124-225b5271-7676-4b35-bf7c-1ad441505fa2.png">

Reply:
Also, to note we already limit it to 4 when editing a panel in the UI:

<img width="269" alt="image" src="https://user-images.githubusercontent.com/9369625/206565192-c8a40553-8255-4604-a980-31a66f60f415.png">

Reply:
nvm saw your comment below, I'm going to change this to toFixed so 100 will be correct 😆